yoti_python_sdk/anchor.py

```diff
@@ -0,0 +1,108 @@
+import asn1
+import datetime
+import OpenSSL
+import yoti_python_sdk.protobuf.v1.common_public_api.signed_timestamp_pb2 as compubapi
+
+from OpenSSL import crypto
+from yoti_python_sdk import config
+
+UNKNOWN_EXTENSION = ""
+SOURCE_EXTENSION = "1.3.6.1.4.1.47127.1.1.1"
+VERIFIER_EXTENSION = "1.3.6.1.4.1.47127.1.1.2"
+
+
+class Anchor:
+    anchor_type = "Unknown"
+    sub_type = ""
+    value = ""
+    signed_timestamp = compubapi.SignedTimestamp()
+
+    def __init__(self, anchor_type=None, sub_type=None, value=None, signed_timestamp=None):
+        self.anchor_type = anchor_type
+        self.sub_type = sub_type
+        self.value = value
+        self.signed_timestamp = signed_timestamp
+
+    def __iter__(self):
+        return self
+
+    def __next__(self):
+        self.idx += 1
+        try:
+            return self.data[self.idx - 1]
+        except IndexError:
+            self.idx = 0
+            raise StopIteration
+
+    next = __next__  # python2.x compatibility.
+
+    @staticmethod
+    def parse_anchors(anchors):
+        for anc in anchors:
+            parsed_anchors = []
+
+            if hasattr(anc, 'origin_server_certs'):
+                origin_server_certs_list = list(anc.origin_server_certs)
+                origin_server_certs_item = origin_server_certs_list[0]
+
+                cert = crypto.load_certificate(OpenSSL.crypto.FILETYPE_ASN1, origin_server_certs_item).to_cryptography()
+
+                for i in range(len(cert.extensions)):
+                    extensions = cert.extensions[i]
+                    if hasattr(extensions, 'oid'):

```

Comment:
this method has a lot of if nesting.
I suggest that you encapsulate the processing of a single extension in a separate method.